contracts/wrapperContracts/WrapFeeBurner.sol

```diff
@@ -0,0 +1,227 @@
+    function setFeeSharingValue(uint feeBps) public onlyAdmin {
+        require(feeBps < 10000);
+        feeSharingBps = feeBps;
+    }
+
+    event WalletRegisteredForFeeSharing(address sender, address walletAddress);
+    function registerWalletForFeeSharing(address walletAddress) public {
+        require(feeBurnerContract.walletFeesInBps(walletAddress) == 0);
+
+        // if fee sharing value is 0. means the wallet wasn't added.
+        feeBurnerContract.setWalletFees(walletAddress, feeSharingBps);
+        feeSharingWallets.push(walletAddress);
+        WalletRegisteredForFeeSharing(msg.sender, walletAddress);
+    }
+
+    function getFeeSharingWallets() public view returns(address[]) {
+        return feeSharingWallets;
+    }
+
+    // knc rate handling
+    //////////////////////
+    function setPendingKNCRateRange(uint minRate, uint maxRate) public onlyOperator {
+        require(minRate < maxRate);
+        require(minRate > 0);
+
+        //update data tracking
+        setNewData(KNC_RATE_RANGE_INDEX);
+
+        kncPerEth.pendingMinRate = minRate;
+        kncPerEth.pendingMaxRate = maxRate;
+    }
+
+    function approveKNCRateRange(uint nonce) public onlyOperator {
+        if (addSignature(KNC_RATE_RANGE_INDEX, nonce, msg.sender)) {
+            // can perform operation.
+            kncPerEth.minRate = kncPerEth.pendingMinRate;
+            kncPerEth.maxRate = kncPerEth.pendingMaxRate;
+        }
+    }
+
+    function getPendingKNCRateRange() public view returns(uint minRate, uint maxRate, uint nonce) {
+        address[] memory signatures;
+        minRate = kncPerEth.pendingMinRate;
+        maxRate = kncPerEth.pendingMaxRate;
+        (signatures, nonce) = getDataTrackingParameters(KNC_RATE_RANGE_INDEX);
+
+        return(minRate, maxRate, nonce);
+    }
+
+    function getKNCRateRange() public view returns(uint minRate, uint maxRate) {
+        minRate = kncPerEth.minRate;
+        maxRate = kncPerEth.maxRate;
+        return(minRate, maxRate);
+    }
+
+    function setKNCPerEthRate(uint kncPerEther) public onlyOperator {
+        require(kncPerEther >= kncPerEth.minRate);
+        require(kncPerEther <= kncPerEth.maxRate);
+        feeBurnerContract.setKNCRate(kncPerEther);
+    }
+
+    function getKNCRateRangeSignatures() public view returns (address[] signatures) {
+        uint nonce;
+        (signatures, nonce) = getDataTrackingParameters(KNC_RATE_RANGE_INDEX);
+        return(signatures);
+    }
+
+    //set reserve data
+    //////////////////
+    function setPendingReserveData(address _reserve, uint feeBps, address kncWallet) public onlyOperator {
+        require(_reserve != address(0));
+        require(kncWallet != address(0));
+        require(feeBps > 0);
+
+        addReserve.reserve = _reserve;
+        addReserve.feeBps = feeBps;
+        addReserve.kncWallet = kncWallet;
+        setNewData(ADD_RESERVE_INDEX);
+    }
+    
+    function approveAddReserveData(uint nonce) public onlyOperator {
```

Comment:
also here - consider moving approveAddReserveData to after getPendingAddReserveData and getAddReserveSignatures.